### PR TITLE
handle document not found error in avatax during cancel

### PR DESCRIPTION
**Pull request by lkostrowski**
## Checklist

- [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

File: apps/avatax/src/pages/api/webhooks/order-cancelled.ts [hunk 1]
```diff
@@ -139,12 +141,22 @@ export default wrapWithLoggerContext(
...
+              providerConfig.value.avataxConfig.config,
+            );
+          } catch (e) {
+            if (e instanceof AvataxOrderCancelledAdapter.DocumentNotFoundError) {
+              logger.warn("Document was not found in AvaTax. Responding 400", {
+                error: e,
+              });
+
+              return res.status(400).end();
+            }
```
Comment (krzysztofzuraw) [inline apps/avatax/src/pages/api/webhooks/order-cancelled.ts:158]: **Question:** what about other type of error? Shoudn't we have here 500 returned and error logged to Sentry?

File: apps/avatax/src/modules/avatax/order-cancelled/avatax-order-cancelled-adapter.ts [hunk 2]
```diff
@@ -11,11 +15,28 @@ export type AvataxOrderCancelledTarget = VoidTransactionArgs;
...
   constructor(
     private avataxClient: AvataxClient,
     private avataxOrderCancelledPayloadTransformer: AvataxOrderCancelledPayloadTransformer,
   ) {}
 
+  /**
+   * Locally extract error code - but this should be more global, with some normalization/parsing logic/middleware
+   */
+  private extractAvaTaxErrorCode = (error: unknown): string | null => {
+    const parsedError = z.object({ code: z.string() }).safeParse(error);
```
Comment (krzysztofzuraw) [inline apps/avatax/src/modules/avatax/order-cancelled/avatax-order-cancelled-adapter.ts:31]: **Question:** can't we have enum here and add support for additional error codes like https://github.com/saleor/apps/blob/main/apps/avatax/src/modules/avatax/avatax-errors-parser.ts ?

File: apps/avatax/src/pages/api/webhooks/order-cancelled.ts (hunk 1 above)
Comment (lkostrowski) [reply apps/avatax/src/pages/api/webhooks/order-cancelled.ts:158]: yea, good point

File: apps/avatax/src/modules/avatax/order-cancelled/avatax-order-cancelled-adapter.ts (hunk 2 above)
Comment (lkostrowski) [reply apps/avatax/src/modules/avatax/order-cancelled/avatax-order-cancelled-adapter.ts:31]: yes!

**Review by krzysztofzuraw — APPROVED** [review]

**Pull request merged**